ledger/complete/mtrie/trie/trie.go

```diff
@@ -375,37 +440,54 @@ func update(
 
 	// recurse over each branch
 	var lChild, rChild *node.Node
+	var lRegCountDelta, rRegCountDelta int64
+	var lRegSizeDelta, rRegSizeDelta int64
+	var lLowestHeightTouched, rLowestHeightTouched int
 	parallelRecursionThreshold := 16
 	if len(lpaths) < parallelRecursionThreshold || len(rpaths) < parallelRecursionThreshold {
 		// runtime optimization: if there are _no_ updates for either left or right sub-tree, proceed single-threaded
-		lChild = update(nodeHeight-1, lchildParent, lpaths, lpayloads, lcompactLeaf, prune)
-		rChild = update(nodeHeight-1, rchildParent, rpaths, rpayloads, rcompactLeaf, prune)
+		lChild, lRegCountDelta, lRegSizeDelta, lLowestHeightTouched = update(nodeHeight-1, lchildParent, lpaths, lpayloads, lcompactLeaf, prune)
+		rChild, rRegCountDelta, rRegSizeDelta, rLowestHeightTouched = update(nodeHeight-1, rchildParent, rpaths, rpayloads, rcompactLeaf, prune)
 	} else {
 		// runtime optimization: process the left child is a separate thread
-		wg := sync.WaitGroup{}
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
-			lChild = update(nodeHeight-1, lchildParent, lpaths, lpayloads, lcompactLeaf, prune)
-		}()
-		rChild = update(nodeHeight-1, rchildParent, rpaths, rpayloads, rcompactLeaf, prune)
-		wg.Wait()
+
+		// Since we're receiving 4 values from goroutine, use a
+		// channel to prevent them going on the heap.
+
```

Comment:
```suggestion
```

Reply:
Fixed in 38a12bc65f9a3f3510e23923c8e1a1a15c3d38cc.